### PR TITLE
KVM nested virtualization (jsc#SLE-11329)

### PR DESCRIPTION
**Pull request by tbazant**
### Description
Adds basic info to enable nested virtualization for KVM

*Are backports required?*

- [x] To maintenance/SLE15SP1
- [x] To maintenance/SLE15SP0
- [x] To maintenance/SLE12SP5
- [x] To maintenance/SLE12SP4
- [x] To maintenance/SLE12SP3

### CODE REVIEW

File: xml/vt_installation.xml [hunk 1]
```diff
@@ -258,4 +258,75 @@
...
+   </varlistentry>
+  </variablelist>
+  <para>
+   Nested virtualization is useful mainly in cloud environments as you can
+   manage your own virtual machines there directly with your favorite
+   hypervisor of choice. It can also enable the live migration of hypervisors
+   and their guest virtual machines as a single entity.
+ </para>
+ <para>
+  To enable nesting, set the <option>nested</option> module parameter to
```
Comment (jfehlig) [inline xml/vt_installation.xml:309]: set the <option>nested</option> KVM module parameter ... ?

In addition to editing /etc/modprobe.d/kvm_intel.conf you can enable nesting more dynamically by removing the module and reloading it with the parameter. E.g. 'modprobe -r kvm_intel; modprobe kvm_intel nested=1'.

File: xml/vt_installation.xml [hunk 2]
```diff
@@ -258,4 +258,75 @@
...
+  <literal>Y</literal>.
+ </para>
+ <itemizedlist>
+  <listitem>
+   <para>
+    For Intel CPUs, edit <filename>/etc/modprobe.d/kvm_intel.conf</filename>
+    and add the following line:
+   </para>
+   <screen>options kvm-intel nested=Y</screen>
```
Comment (jfehlig) [inline xml/vt_installation.xml:318]: s/kvm-intel/kvm_intel/

File: xml/vt_installation.xml [hunk 3]
```diff
@@ -258,4 +258,75 @@
...
+  </listitem><listitem>
+   <para>
+    For AMD CPUs, edit <filename>/etc/modprobe.d/kvm_amd.conf</filename>
+    and add the following line:
+   </para>
+   <screen>options kvm-amd nested=Y</screen>
```
Comment (jfehlig) [inline xml/vt_installation.xml:324]: Same there. The module name is kvm_amd.

File: xml/vt_installation.xml [hunk 4]
```diff
@@ -258,4 +258,75 @@
...
+    <listitem>
+     <para>
+      A virtual machine running on L1. It is called a <emphasis>nested
+       guest</emphasis>.
+     </para>
+    </listitem>
+   </varlistentry>
+  </variablelist>
+  <para>
+   Nested virtualization is useful mainly in cloud environments as you can
```
Comment (jfehlig) [inline xml/vt_installation.xml:303]: Running your own hypervisor in a VM in the cloud is definitely one use-case, but dev/testing is another. Also I'd suggest dropping 'favorite' from this sentence and just go with "your hypervisor of choice".

File: xml/vt_installation.xml [hunk 5]
```diff
@@ -258,4 +258,75 @@
...
+   <para>
+    For AMD CPUs, edit <filename>/etc/modprobe.d/kvm_amd.conf</filename>
+    and add the following line:
+   </para>
+   <screen>options kvm-amd nested=Y</screen>
+  </listitem>
+ </itemizedlist>
+ <para>
+  When your L0 host is capable of nesting, you will be able to start an L1
+  guest with the <option>-cpu host</option> option.
```
Comment (bfrogers) [inline xml/vt_installation.xml:329]: ... guest with the <option>-cpu host</option> QEMU command-line option, or by adding the vmx (for Intel CPUs) or the svm  (for AMD CPUs) cpu feature to the -cpu <model> QEMU command-line option.

(or words to that effect).

File: xml/vt_installation.xml [hunk 6]
```diff
@@ -258,4 +258,75 @@
...
+     <para>
+      A virtual machine running on L1. It is called a <emphasis>nested
+       guest</emphasis>.
+     </para>
+    </listitem>
+   </varlistentry>
+  </variablelist>
+  <para>
+   Nested virtualization is useful mainly in cloud environments as you can
+   manage your own virtual machines there directly with your favorite
```
Comment (lproven) [inline xml/vt_installation.xml]:
```suggestion
   manage your own virtual machines there directly with your
```

Comment (lproven) [reply xml/vt_installation.xml]: "Favourite hypervisor of choice" seems redundant.

**Pull request merged**